x-pack/plugins/cases/common/api/cases/comment/files.ts

```diff
@@ -9,19 +9,21 @@ import * as rt from 'io-ts';
 import { MAX_DELETE_FILES } from '../../../constants';
 import { limitedArraySchema, NonEmptyString } from '../../../schema';
 
+export const SingleFileAttachmentMetadataRt = rt.type({
+  name: rt.string,
+  extension: rt.string,
+  mimeType: rt.string,
+  created: rt.string,
+});
+
 export const FileAttachmentMetadataRt = rt.type({
-  files: rt.array(
-    rt.type({
-      name: rt.string,
-      extension: rt.string,
-      mimeType: rt.string,
-      created: rt.string,
-    })
-  ),
+  files: rt.array(SingleFileAttachmentMetadataRt),
 });
 
 export type FileAttachmentMetadata = rt.TypeOf<typeof FileAttachmentMetadataRt>;
 
+export type DownloadableFile = rt.TypeOf<typeof SingleFileAttachmentMetadataRt> & { id: string };
```

Comment:
Thanks for refactoring these types! I think we've generally kept the files in `api` for io-ts types. How about we move this type to a new type file in `cases/public/components/files/types.ts`?

@cnasikas what do you think?

Reply:
Yes if it is used (`DownloadableFile`) only by the `public` is it better to be in `cases/public/components/files/types.ts`.